### ISSUE TITLE
ios Ionic Cordova

### ISSUE THREAD

**Issue by asa47** [markmarijnissen/cordova-app-loader#82]
Hello, I have been trying to add cordova app loader in our ios app, unfortunately I could not make it work;
probably I am doing something wrong or I might missing something....

So far it works great on the browser(chrome) however once i build it on ios, it just does not work, and when ever I call the function loader.check() nothing happens. 

Even with `loader.check().then(function(){console.log('success');}, function(){console.log('fail');})`, it does not print any output.

Any advice?
Regards

**Comment by ostruk** [comment]
I had a similar issue and I found that the problem was with the order of how ios executes code when promises get resolved immediately. Place console.log throughout the bootstrapping process of your app and compare the order of execution between android and ios. Specifically look for differences around key points like loading of the manifest, initialization of the filesystem and initialization of the CordovaAppLoader.

Also try to place debugging code into the library itself or just step through the lines to see why it's failing internally, I found that code not difficult to understand.

Good luck!
